### PR TITLE
Use NettyClientLogger everywhere

### PR DESCRIPTION
**Pull request by dagnir**
## Motivation and Context
Improved logging.

## Description
Using `NettyClientLogger` everywhere in the Netty client helps ensure we log the associated `Channel` details when available and keep that logging uniform throughout.

## Testing
New unit tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have read the **README** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java [hunk 1]
```diff
@@ -69,7 +68,7 @@
 
     private static final String CLIENT_NAME = "NettyNio";
 
-    private static final Logger log = LoggerFactory.getLogger(NettyNioAsyncHttpClient.class);
+    private static final NettyClientLogger log = NettyClientLogger.getLogger(NettyNioAsyncHttpClient.class);
```
Comment (joviegas) [inline http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java:71]: Could you please help me to confirm if there will be no changes in the way the User can enable Debug logs as mentioned in https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/logging-slf4j.html ?
apart from adding 
`log4j.logger.io.netty.handler.logging.LoggingHandler=DEBUG`
If there is a difference in how user can enable the logs now ,can we document this in this PR  ?

File: http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java [hunk 2]
```diff
@@ -55,6 +53,7 @@
 import software.amazon.awssdk.http.nio.netty.internal.SdkChannelPool;
 import software.amazon.awssdk.http.nio.netty.internal.SdkChannelPoolMap;
 import software.amazon.awssdk.http.nio.netty.internal.SharedSdkEventLoopGroup;
+import software.amazon.awssdk.http.nio.netty.internal.utils.NettyClientLogger;
```
Comment (joviegas) [inline http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java:56]: Should we also mention this changes in changelogs ?

Comment (dagnir) [reply http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java:56]: I don't think so, this is an implementation/internal detail

File: http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java (hunk 1 above)
Comment (dagnir) [reply http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java:71]: No it doesn't change those details. The logger name is based on the class name given to the `getLogger` method, which stays the same.

**Review by joviegas — APPROVED** [review]

**Comment by dagnir** [comment]
Coverage is low because this change updates logging lines that were previously uncovered.

**Pull request merged**